### PR TITLE
Feature laser turret logic

### PR DESCRIPTION
**Pull request by Slimmerik**

### CODE REVIEW

File: code/headers/hc_sr04.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+    /**
+ *  \brief
+ *  This class is for reading the data from the HC-SR04 ultrasone sensor
+ *
+ *  \details
+ *   This class return the distance measured with the HC-SR04 ultrasone sensor, accurate up to 400 centimeters.
+ *   After 400 centimeters measurements are unuseable and should be disgarded.
+ */
+    class hc_sr04_c
```
Comment (Joepieler) [inline code/headers/hc_sr04.hpp]: Wasn't an ultrasonic sensor already written?

File: code/headers/laser.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#include <hwlib.hpp>
+
+namespace r2d2::laser {
+    class laser_c {
+    public:
+        /**
+         * The constructor of the laser object
+         *
+         */
+        laser_c(uint8_t gpio);
```
Comment (Joepieler) [inline code/headers/laser.hpp]: why not a hwlib pin_out?

File: code/headers/laser_turret_controller.hpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+         * @param x_axis_servo adds object to control the x axis servo
+         * @param y_axis_servo adds object to control the y axis servo
+         */
+        laser_turret_controller_c(r2d2::sg90::sg90_c &x_axis_servo
+                ,r2d2::sg90::sg90_c &y_axis_servo
+                ,r2d2::laser::laser_c &laser
+                ,r2d2::distance::hc_sr04_c &distance_sensor);
+
+        /**
+         * This methode updates the point the laser has to point at and also takes the distance in to account.
```
Comment (Joepieler) [inline code/headers/laser_turret_controller.hpp]: typo must be method same for line 44 and 52

File: code/headers/hc_sr04.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private:
+        hwlib::pin_in & echo_pin;       //Listens to a high value from returning ultrasone sounds
+        hwlib::pin_out & trigger_pin;   //Triggers the sensor to send out a ultrasone frequency
+    public:
+        hc_sr04_c(hwlib::pin_in & echo_pin, hwlib::pin_out & trigger_pin);
+
+        /**
+        * return measured distance in centimeters.
+        */
+        int16_t get_distance();
```
Comment (Joepieler) [inline code/headers/hc_sr04.hpp]: can de value be below zero? if not that is should be uint16_t

File: code/headers/sg90.hpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+         * @param degrees This parameter
+         */
+        void move_servo_to_degrees(uint16_t degrees);
+
+        /**
+         * This method returns the position of the servo in degrees by taking the duty cycle stop form pca9685.
+         *
+         * @return returns the position of the servo in degrees.
+         */
+        uint16_t get_servo_position();
```
Comment (Joepieler) [inline code/headers/sg90.hpp]: can the servo turn more than 255 degrees? i think you can use a uint8_t

File: code/src/hc_sr04.cpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    uint32_t ticks_start = hwlib::now_ticks();
+
+    while(echo_pin.read()){
+        hwlib::wait_us(5);
+    }
+
+    uint32_t tick_counter = hwlib::now_ticks() - ticks_start;
+    tick_counter = tick_counter / (hwlib::ticks_per_us() * 2) * 0.035 * 10; // formula to change ticks into centimeters.
+    // if(tick_counter > 399) {
```
Comment (Joepieler) [inline code/src/hc_sr04.cpp]: can be removed?

File: code/headers/sg90.hpp (hunk 5 above)
Comment (Slimmerik) [reply code/headers/sg90.hpp]: range is 0-180 but to give the servo more steps the input/output value will was multiplied by 10.

File: code/headers/hc_sr04.hpp (hunk 4 above)
Comment (Slimmerik) [reply code/headers/hc_sr04.hpp]: i took the code from the r2d2 project and didn't look at the workings of. only edited the naming to be the same as in the guidelines.

File: code/headers/laser_turret_controller.hpp (hunk 3 above)
Comment (Slimmerik) [reply code/headers/laser_turret_controller.hpp]: i indeed took the code from the r2d2 project.

**Pull request merged**